src/stdlib/zip.rs

```diff
@@ -0,0 +1,125 @@
+use crate::compiler::prelude::*;
+
+fn zip(value: Value) -> Resolved {
+    Ok(MultiZip(
+        value
+            .try_array()?
+            .into_iter()
+            .map(|value| value.try_array().map(Vec::into_iter))
+            .collect::<Result<_, _>>()?,
+    )
+    .collect::<Vec<_>>()
+    .into())
+}
+
+struct MultiZip(Vec<std::vec::IntoIter<Value>>);
+
+impl Iterator for MultiZip {
+    type Item = Vec<Value>;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.iter_mut().map(Iterator::next).collect()
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Zip;
+
+impl Function for Zip {
+    fn identifier(&self) -> &'static str {
+        "zip"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "array",
+            kind: kind::ARRAY,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "merge three arrays into an array of 3-tuples",
+            source: r#"zip([["a", "b", "c"], [1, null, true], [4, 5, 6]])"#,
+            result: Ok(r#"[["a", 1, 4], ["b", null, 5], ["c", true, 6]]"#),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let array = arguments.required("array");
+        Ok(ZipFn { array }.as_expr())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ZipFn {
+    array: Box<dyn Expression>,
+}
+
+impl FunctionExpression for ZipFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        zip(self.array.resolve(ctx)?)
+    }
+
+    fn type_def(&self, _state: &TypeState) -> TypeDef {
+        TypeDef::object(Collection::any())
```

Comment:
Should this be `array` instead of `object`?

Reply:
Of course, yes. Fixed in d703cd532